Source/SPConnectionController.m

```diff
@@ -3335,6 +3415,10 @@ - (id)initWithDocument:(SPDatabaseDocument *)document
 		[socketColorField setColorList:colorList];
 		[socketColorField   bind:@"selectedTag" toObject:self withKeyPath:@"colorIndex" options:nil];
 
+		standardTimeZoneField.menu.itemArray = [self generateTimeZoneMenuItems];
+		sshTimeZoneField.menu.itemArray = [self generateTimeZoneMenuItems];
+		socketTimeZoneField.menu.itemArray = [self generateTimeZoneMenuItems];
+
```

Comment:
Can we instead generate the array, store it in a temp var, and use it across all three connections, instead of generating the items list 3x?

Reply:
When I tried that it complained that I was using the same menu items in multiple menus. But I checked the docs right now and the property has the `copy` attribute so that shouldn't have been a problem. I can try again.

Reply:
Tried again. I forgot that `copy` just means that the NSArray instance is getting copied, but the copy will still contain the same instances of `NSMenuItem`. Also, the setter for `itemArray` is just a wrapper around a loop of `insertItem:atIndex`. I'll add a comment to the code.